src/main/java/org/jenkins/tools/test/hook/MultiParentCompileHook.java

```diff
@@ -132,15 +127,15 @@ private void compile(
             File localCheckoutDir,
             String parentFolder,
             String pluginName)
-            throws PomExecutionException, IOException {
+            throws PomExecutionException {
         if (isSnapshotMultiParentPlugin(parentFolder, path, localCheckoutDir)) {
             // "process-test-classes" not working properly on multi-module plugin.
             // See https://issues.jenkins.io/browse/JENKINS-62658
             // installs dependencies into local repository
             String mavenModule =
                     PluginCompatTester.getMavenModule(pluginName, path, runner, mavenConfig);
             if (mavenModule == null || mavenModule.isBlank()) {
-                throw new IOException(
+                throw new IllegalStateException(
```

Comment:
Yet more abuse of `IOException` for things that aren't I/O errors. Picked a more appropriate exception.